shine/scene.py

```diff
@@ -0,0 +1,82 @@
+import numpyro
+import numpyro.distributions as dist
+import jax.numpy as jnp
+import jax
+import jax_galsim as galsim
+from shine.config import ShineConfig, DistributionConfig
+
+class SceneBuilder:
+    def __init__(self, config: ShineConfig):
+        self.config = config
+
+    def _parse_prior(self, name: str, param_config):
+        """Helper to create NumPyro distributions from config or return fixed value."""
+        if isinstance(param_config, (float, int)):
+            return float(param_config)
+        
+        if isinstance(param_config, DistributionConfig):
+            if param_config.type == 'Normal':
+                return numpyro.sample(name, dist.Normal(param_config.mean, param_config.sigma))
+            elif param_config.type == 'LogNormal':
+                 return numpyro.sample(name, dist.LogNormal(jnp.log(param_config.mean), param_config.sigma))
+            elif param_config.type == 'Uniform':
+                return numpyro.sample(name, dist.Uniform(param_config.min, param_config.max))
+            else:
+                raise ValueError(f"Unknown distribution type: {param_config.type}")
+        
+        return param_config
+
+    def build_model(self):
+        """Returns a callable model function for NumPyro."""
+        
+        def model(observed_data=None, psf_config=None):
+            # Define GSParams with fixed FFT size to avoid dynamic shape issues in JAX
+            fft_size = 128
+            gsparams = galsim.GSParams(maximum_fft_size=fft_size, minimum_fft_size=fft_size)
+
+            # --- 0. Build PSF from config using JAX-GalSim ---
+            if psf_config['type'] == 'Gaussian':
+                psf = galsim.Gaussian(sigma=psf_config['sigma'], gsparams=gsparams)
+            else:
+                raise NotImplementedError(f"PSF type {psf_config['type']} not implemented")
+            
+            # --- 1. Global Parameters (Shear) ---
+            g1 = self._parse_prior("g1", self.config.gal.shear.g1)
+            g2 = self._parse_prior("g2", self.config.gal.shear.g2)
+            shear = galsim.Shear(g1=g1, g2=g2)
+            
+            # --- 2. Galaxy Population ---
+            n_galaxies = self.config.image.n_objects
+            
+            with numpyro.plate("galaxies", n_galaxies):
+                flux = self._parse_prior("flux", self.config.gal.flux)
+                hlr = self._parse_prior("hlr", self.config.gal.half_light_radius)
+                
+                x = self.config.image.size_x / 2.0
```

Comment:
Here we are fixing all the galaxies at the centre of the image. This is ok if we were dealing with individual postage stamps, but at the end of the function it adds all the images together as if we were simulating a single wider field.